src/Application/ItemTypeLabelLookup.php

```diff
@@ -4,10 +4,11 @@
 
 namespace ProfessionalWiki\WikibaseFacetedSearch\Application;
 
-use Wikibase\DataModel\Entity\ItemId;
+use Wikibase\DataModel\Entity\EntityId;
 
+// TODO: Make into a generic class that can be used for other labels
 interface ItemTypeLabelLookup {
 
-	public function getLabel( ItemId $itemType ): string;
+	public function getLabel( EntityId $itemType ): string;
```

Comment:
This interface is designed for its existing consumer. Using it and its implementation of `FallbackItemTypeLabelLookup` elsewhere for other use cases (generic Wikibase Item label lookup and Property label lookup) isn't good.